### PR TITLE
Added NSFW Commands

### PR DESCRIPTION
**Pull request by GrandMoff100**

### CODE REVIEW

**Comment by GrandMoff100** [comment]
Doesn't contain docstrings.

File: bot/cogs/nsfw/neko.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+import discord
+import requests
+from discord.ext import commands
+
+
+async def make_embed(img: str, provider: str, author: discord.User):
+    em = discord.Embed(color=discord.Colour.red())
+    em.title = provider + " requested by " + author.name
+    em.set_image(url=img)
+    return em
```
Comment (janaSunrise) [inline bot/cogs/nsfw/neko.py:12]:
```suggestion
    embed = discord.Embed(color=discord.Colour.red())
    embed.title = provider + " requested by " + author.name
    embed.set_image(url=img)
    return embed
```

File: bot/cogs/nsfw/neko.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    def __init__(self, bot):
+        self.bot = bot
+        self.config = bot.config
+
+    async def get(self, url, author: discord.User):
+        base = 'https://api.nekos.dev/api/v3/'
+        r = requests.get(base + url)
+        r = r.json()
+        em = await make_embed(r['data']['response']["url"], "Nekos.life", author)
+        return em
```
Comment (janaSunrise) [inline bot/cogs/nsfw/neko.py:27]:
```suggestion
        req = requests.get(base + url)
        req = req.json()
        embed = await make_embed(req['data']['response']["url"], "Nekos.life", author)
        return embed
```

File: bot/cogs/nsfw/neko.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+            await ctx.invoke(self.sfw_neko)
+
+    @neko.command(name='neko')
+    @commands.guild_only()
+    async def sfw_neko(self, ctx):
+        async with ctx.typing():
+            sources = ["images/sfw/img/neko", "images/sfw/gif/neko"]
+            source = choice(sources)
+            em = await self.get(source, ctx.author)
+            msg: discord.Message = await ctx.send(embed=em)
```
Comment (janaSunrise) [inline bot/cogs/nsfw/neko.py:43]:
```suggestion
            embed = await self.get(source, ctx.author)
            msg = await ctx.send(embed=embed)
```

File: bot/cogs/nsfw/neko.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+            await msg.add_reaction("😻")
```
Comment (janaSunrise) [inline bot/cogs/nsfw/neko.py:44]:
```suggestion
```

File: bot/cogs/nsfw/neko.py [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+    @neko.command(name='nsfw')
+    @commands.guild_only()
+    @commands.is_nsfw()
+    async def nsfw_neko(self, ctx):
+        sources = ["images/nsfw/gif/neko", "images/nsfw/img/neko_lewd", "images/nsfw/img/neko_ero"]
+        source = choice(sources)
+        em = await self.get(source, ctx.author)
+        msg = await ctx.send(embed=em)
+        await msg.add_reaction("😻")
```
Comment (janaSunrise) [inline bot/cogs/nsfw/neko.py:54]:
```suggestion
        embed = await self.get(source, ctx.author)
        msg = await ctx.send(embed=embed)
```

File: bot/cogs/nsfw/neko.py [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+
+    @neko.command(name="waifu")
+    @commands.guild_only()
+    async def sfw_waifu(self, ctx):
+        async with ctx.typing():
+            source = "images/sfw/img/waifu"
+            em = await self.get(source, ctx.author)
+            msg: discord.Message = await ctx.send(embed=em)
+            await msg.add_reaction("❤️")
```
Comment (janaSunrise) [inline bot/cogs/nsfw/neko.py:63]:
```suggestion
            embed = await self.get(source, ctx.author)
            msg = await ctx.send(embed=embed)
```

File: bot/cogs/nsfw/neko.py [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+
+    @neko.command(name="kitsune")
+    @commands.guild_only()
+    async def sfw_kitsune(self, ctx):
+        async with ctx.typing():
+            source = "images/sfw/img/kitsune"
+            em = await self.get(source, ctx.author)
+            msg: discord.Message = await ctx.send(embed=em)
+            await msg.add_reaction("❤️")
```
Comment (janaSunrise) [inline bot/cogs/nsfw/neko.py:72]:
```suggestion
    @neko.command(name="kitsune")
    @commands.guild_only()
    async def sfw_kitsune(self, ctx):
        async with ctx.typing():
            source = "images/sfw/img/kitsune"
            embed = await self.get(source, ctx.author)
            msg = await ctx.send(embed=embed)
```

File: bot/cogs/nsfw/neko.py [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+
+    @neko.command(name="lewd")
+    @commands.guild_only()
+    @commands.is_nsfw()
+    async def nsfw_classic(self, ctx):
+        sources = ["images/nsfw/img/classic_lewd", "images/nsfw/img/neko_lewd", "images/nsfw/img/neko_ero"]
+        source = choice(sources)
+        em = await self.get(source, ctx.author)
+        msg: discord.Message = await ctx.send(embed=em)
+        await msg.add_reaction("😻")
```
Comment (janaSunrise) [inline bot/cogs/nsfw/neko.py:82]:
```suggestion
        embed = await self.get(source, ctx.author)
        msg = await ctx.send(embed=embed)
```

File: bot/cogs/nsfw/neko.py [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+
+    @neko.command(name="blowjob")
+    @commands.guild_only()
+    @commands.is_nsfw()
+    async def nsfw_blowjob(self, ctx):
+        sources = ["images/nsfw/gif/blow_job", "images/nsfw/img/blowjob_lewd"]
+        source = choice(sources)
+        em = await self.get(source, ctx.author)
+        msg: discord.Message = await ctx.send(embed=em)
+        await msg.add_reaction("❤️")
```
Comment (janaSunrise) [inline bot/cogs/nsfw/neko.py:92]:
```suggestion
        embed = await self.get(source, ctx.author)
        msg = await ctx.send(embed=embed)
```

File: bot/cogs/nsfw/neko.py [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+
+    @neko.command(name="furry")
+    @commands.guild_only()
+    @commands.is_nsfw()
+    async def nsfw_furry(self, ctx):
+        sources = ["images/nsfw/gif/yiff", "images/nsfw/img/yiff_lewd"]
+        source = choice(sources)
+        em = await self.get(source, ctx.author)
+        msg: discord.Message = await ctx.send(embed=em)
+        await msg.add_reaction("❤️")
```
Comment (janaSunrise) [inline bot/cogs/nsfw/neko.py:102]:
```suggestion
        embed = await self.get(source, ctx.author)
        msg = await ctx.send(embed=embed)
```

**Review by janaSunrise — CHANGES_REQUESTED** [review]
Please request those changes, and edit the files to use proper naming convention, and try to make it different, so it clearer

**Review by ItsDrike — CHANGES_REQUESTED** [review]
While I've requested some changes to code itself isn't that bad, although I do want to state my position on this which is that I'm personally not a huge fan of having NSFW commands like this in the bot and after these changes will get resolved I won't approve this PR because of that if it will get 2 approvals anyway, it can be merged even though I'm personally against having this in our bot.

File: bot/cogs/reddit.py [hunk 11]
```diff
@@ -62,7 +91,6 @@ def __init__(self, bot: Bot) -> None:
     @group()
     async def reddit(self, ctx: Context) -> None:
         """Reddit commands."""
-        pass
```
[the remark this answers is not in the document]
Comment (janaSunrise) [reply bot/cogs/reddit.py:65]: i got this error from flake8-bear

[the remark this answers is not in the document]
Comment (ItsDrike) [reply bot/cogs/reddit.py:65]: Can you show me the specific error number with corresponding message to it?

**Pull request merged**